packages/grid/_modules_/grid/hooks/features/pagination/gridPaginationReducer.ts

```diff
@@ -88,7 +88,7 @@ export const setGridRowCountStateUpdate = (state, payload): GridPaginationState
       ...state,
       pageCount: newPageCount,
       rowCount: totalRowCount,
-      page: state.page > newPageCount ? newPageCount : state.page,
+      page: state.page > newPageCount ? 0 : state.page,
```

Comment:
if `state.page > newPageCount` then users are looking at the last page and when they apply a filter, they should probably not go back to the first page, but stay on the last page of the filtered rows. 
Similarly, if there is an update that delete rows due to streaming, users should not jump back to the first page.

As page is 0 based, have you tried.
```tsx
page: state.page > newPageCount ? newPageCount - 1  : state.page,
```

Reply:
> they should probably not go back to the first page, but stay on the last page of the filtered rows.

I assumed it should go back to first page. I also had thought of this solution you mentioned. It works.
```js
page: state.page > newPageCount ? newPageCount - 1  : state.page
```
Saw later in the expected results of the issue and it is to stay on the last page of filtered rows 👍 
![Screenshot 2021-05-06 at 4 43 24 PM](https://user-images.githubusercontent.com/20900032/117290030-f12e5f00-ae8a-11eb-82a2-ae81c76ef704.png)

Updated.

Reply:
> I assumed it should go back to first page

👍 for going to the first page when the filter change, but only when the filter change.

Reply:
> 👍 for going to the first page when the filter change, but only when the filter change.

Yeah, I thought it makes sense to go back to first page. But @dtassone mentions the below use case:
>  if there is an update that delete rows due to streaming

Reply:
@ZeeshanTamboli The use case the Damien has raised is unrelated to filtering.

Reply:
> 👍 for going to the first page when the filter change, but only when the filter change. From a UX perspective, it seems to make the most sense. But maybe if you are already on the last page, you will want to stay on the last page (we could do an exception for the last page).

Not up for the exception! It's one or the other.

I think if you go back to the first page every time you change the filter (so on keydown), it would be a bad UX.

Reply:
> I think if you go back to the first page every time you change the filter (so on keydown), it would be a bad UX.

From my perspective, if you are on page 4 and change the filter, the 4 previous pages, are now completely unknown. I would personally go back to page 0 each time to retraverse the list. This being done automatically seems better. It's what https://demos.telerik.com/kendo-ui/grid/filter-row do. 

However, these keep you on the same page:

- https://ag-grid.com/react-grid/row-pagination/
- https://devexpress.github.io/devextreme-reactive/react/grid/demos/featured/integrated-data-shaping/

Reply:
Imagine you look for a specific Desk in commodity demo.

You start typing `D-12`, you go next, next, next page...
You still can't find it

So you add `3` => If you go back to 0 then you have to go next next and next..., super annoying as you loose track of  where you were!

Reply:
> super annoying as you loose track of where you were!

I don't get this part. If a end-user change the filter, it's because he can't find what he's looking for, correct? If this changes the rows, then by definition, he already does no longer knows where he's after.

Reply:
Airtable doesn't have pagination but we can apply the same reasoning on how they handle the scrollbar. They keep the same scroll-top. So I guess we shouldn't reset the pagination, but only make sure we stay on the same page, if possible.

Reply:
Another argument is that a grid is a tool for analytics. Filtering and pagination are 2 separate features.
And ppl use the grid in different manners, so our goal is to let users use the tool as they want but keep a rational state. 
Changing pages while filtering would make us think and solutionize instead of users. 
We provide the tool, not the solution.

Also would you scroll back up if you change the filter? I wouldn't for the same reasons, I would update the size of the scroll thumb but keep the position

Reply:
Yeah, for me selling part is Airtable. They have likely smart people that have spend hours thinking about this very problem, and likely tried different solutions, so happy to reproduce what they have learned.

Reply:
Ag and airtable should be our top benchmark solutions ;)